### PR TITLE
Add max suggestions prop

### PR DESCRIPTION
**Pull request by sfarberov**
Adds an optional upper bound on the number of suggestions displayed

What did you change (functionally and technically)?

This change provides a prop `maxSuggestions` which enables limiting the number of rendered suggestions to the given maximum.
It also includes an update to the README to document the new prop, as well as a test covering the functionality.

**Checklist** (remove this list before you submit the PR)
* Are there tests for the new code? ✅ 
* Does the code comply to our code conventions? ✅ 
* Does the PR resolve the whole issue? ✅ 

**Additional review hints** (remove this list before you submit the PR)
* Besides the code review, what should the reviewer test?
~When I ran the test suite, there were a number of failing tests on master. I removed the Enzyme `attachTo` option which fixed all tests, but I didn't really investigate the relevant comment regarding why this was needed.~ Undid this change post-rebase with origin/master

* Are there any edge cases?
No

* Do you have any test files or test set-up?
Yes, added to an existing file in the suite.

* Could your changes cause side effects elsewhere in the code base?
Not AFAIK

### CODE REVIEW

File: src/utils/countSuggestions.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
...
-  Object.values(suggestions).reduce(
-    (acc, { results }) => acc + results.length,
-    0
-  )
+import { isUndefined } from 'lodash'
+
+const countSuggestions = (suggestions, maxSuggestions) =>
+  isUndefined(maxSuggestions)
+    ? Object.values(suggestions).reduce((acc, { results }) => acc + results.length, 0)
+    : maxSuggestions
```
Comment (jfschwarz) [inline src/utils/countSuggestions.js]: I believe you should go for a `Math.min()` here. 

Currently, there will be a problem if `maxSuggestions` is defined but there are fewer suggestions than the maximum. In this case, the function won't return the actual count, but the maximum, which will break keyboard navigation again.

(We really need to increase test coverage of this package!)

File: src/MentionsInput.js [hunk 2]
```diff
@@ -91,7 +91,13 @@ const propTypes = {
           : PropTypes.instanceOf(Element),
     }),
   ]),
-
+  maxSuggestions: (props, propName, componentName) => {
+    if (props[propName] < 1) {
```
Comment (jfschwarz) [inline src/MentionsInput.js]: maybe add a check for `typeof props[propName] === 'number'` first?
